aiida/common/progress_reporter.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+# pylint: disable=global-statement,unused-argument
+"""Provide a singleton progress reporter implementation.
+
+The interface is inspired by `tqdm <https://github.com/tqdm/tqdm>`,
+and indeed a valid implementation is::
+
+    from tqdm import tqdm
+    set_progress_reporter(tqdm, bar_format='{l_bar}{bar}{r_bar}')
+
+"""
+from contextlib import contextmanager
+from functools import partial
+from typing import Any, Callable, ContextManager, Iterator, Optional
+
+__all__ = ('get_progress_reporter', 'set_progress_reporter', 'progress_reporter_base', 'ProgressIncrementerBase')
+
+
+class ProgressIncrementerBase:
+    """A base class for incrementing a progress reporter."""
+
+    def set_description_str(self, text: Optional[str] = None, refresh: bool = True):
+        """Set the text shown by the progress reporter.
+
+        :param text: The text to show
+        :param refresh: Force refresh of the progress reporter
+
+        """
+
+    def update(self, n: int = 1):  # pylint: disable=invalid-name
+        """Update the progress counter.
+
+        :param n: Increment to add to the internal counter of iterations
+
+        """
+
+
+@contextmanager
+def progress_reporter_base(*,
+                           total: int,
+                           desc: Optional[str] = None,
+                           **kwargs: Any) -> Iterator[ProgressIncrementerBase]:
+    """A context manager for providing a progress reporter for a process.
```

Comment:
Can you add here that this progress reporter is a dummy that does not print anything?